.github/templates/index.html

```diff
@@ -70,8 +70,18 @@
 <body>
 <div class="container">
     <h1>DANDI Datasets</h1>
+
+    <div class="toc">
+        <h2>Table of Contents</h2>
+        <ul>
+            {% for dandiset in dandisets %}
+            <li><a href="#dandiset-{{ dandiset.id }}">{{ dandiset.metadata.name }}</a></li>
```

Comment:
```suggestion
            <li><a href="#dandiset-{{ dandiset.id }}">{{ dandiset.id }} - {{ dandiset.metadata.name }}</a></li>
```
it alike, since quite often ppl might right away know the id of interest